pkg/volume/iscsi/iscsi_util.go

```diff
@@ -30,9 +31,18 @@ import (
 )
 
 // stat a path, if not exists, retry maxRetries times
-func waitForPathToExist(devicePath string, maxRetries int) bool {
+// when iscsi transports other than default are used,  use glob instead as pci id of device is unknown
+func waitForPathToExist(devicePath string, maxRetries int, deviceIface string) bool {
 	for i := 0; i < maxRetries; i++ {
-		_, err := os.Stat(devicePath)
+		var err error
+		if deviceIface == "default" {
+			_, err = os.Stat(devicePath)
+		} else {
+			fpath, _ := filepath.Glob(devicePath)
```

Comment:
I assume that would require mocking os.Stat & filepath.Glob right ? Could you point me to an example of something similar in the kuberenetes code, can't seem to find much obvious mocking happening.